File: cpp/examples/linux/imu/MPU9250/minimal/main.cpp
#include <cstdio>
#include <cstdlib>
#include <unistd.h>
#include "I2CConnectionLinux.h"
#include "MPU9250.h"

int main() {
    const char* bus_env  = getenv("I2C_BUS");
    const char* addr_env = getenv("I2C_ADDR");
    int     bus  = bus_env  ? atoi(bus_env)       : 1;
    uint8_t addr = addr_env ? (uint8_t)strtol(addr_env, nullptr, 0) : 0x68;
    I2CConnectionLinux connection(bus, addr);

    MPU9250Minimal imu(connection);                                       // Create MPU9250 driver, (connection)

    while (true) {
        float ax, ay, az, gx, gy, gz;
        imu.accel(ax, ay, az);                                            // Read 3-axis acceleration, (float&, float&, float&) → void m/s²
        imu.gyro(gx, gy, gz);                                             // Read 3-axis angular rate, (float&, float&, float&) → void rad/s
        printf("a=%.3f,%.3f,%.3f  g=%.3f,%.3f,%.3f\n", ax, ay, az, gx, gy, gz);
        usleep(100000);
    }
    return 0;
}

File: rust/examples/linux/imu/mpu9250/minimal/src/main.rs
use linux_embedded_hal::{Delay, I2cdev};
use periph::chips::imu::Mpu9250Minimal;

fn main() {
    let i2c_bus: u8 = std::env::var("I2C_BUS").ok().and_then(|v| v.parse().ok()).unwrap_or(1);
    let addr: u8 = std::env::var("I2C_ADDR")
        .ok()
        .and_then(|v| u8::from_str_radix(v.trim_start_matches("0x"), 16).ok())
        .unwrap_or(0x68);

    let dev = I2cdev::new(format!("/dev/i2c-{}", i2c_bus)).expect("open i2c bus");
    let mut delay = Delay;
    let mut imu = Mpu9250Minimal::new(dev, addr, &mut delay).expect("init");

    loop {
        let (ax, ay, az) = imu.accel().expect("read accel");
        let (gx, gy, gz) = imu.gyro().expect("read gyro");
        println!("accel: {:.2} {:.2} {:.2}  gyro: {:.2} {:.2} {:.2}", ax, ay, az, gx, gy, gz);
        std::thread::sleep(std::time::Duration::from_millis(100));
    }
}